### PR TITLE
sys-apps/ethq: new ebuild

### PR DESCRIPTION
**Pull request by SpiderX**
Displays an auto-updating per-second count of the number
of packets and bytes being handled by each queue on
a multi-queue NIC.

Package-Manager: Portage-2.3.49, Repoman-2.3.10

### CODE REVIEW

File: sys-apps/ethq/ethq-0.5.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,52 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+MY_PV=$(ver_rs 1-2 '_')
```
Comment (mgorny) [inline sys-apps/ethq/ethq-0.5.0.ebuild]: Doesn't that replace all separators? Why not just `${PV//./_}` then?

File: sys-apps/ethq/ethq-0.5.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+inherit toolchain-funcs
+
+DESCRIPTION="Ethernet NIC Queue stats viewer"
+HOMEPAGE="https://github.com/isc-projects/ethq"
+SRC_URI="https://github.com/isc-projects/ethq/archive/v${MY_PV}.tar.gz -> ${P}.tar.gz"
```
Comment (mgorny) [inline sys-apps/ethq/ethq-0.5.0.ebuild]: Use `MY_PV` in the filename too.

File: sys-apps/ethq/ethq-0.5.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+DEPEND="sys-libs/ncurses:0"
+RDEPEND="${DEPEND}"
+
+S="${WORKDIR}/${PN}-${MY_PV}"
+
+src_prepare() {
+	default
+
+	# Respect FLAGS
+	sed -i -e '/CXXFLAGS\t/s/= -O3/+=/' \
```
Comment (mgorny) [inline sys-apps/ethq/ethq-0.5.0.ebuild]: I would advise against requiring specific whitespace.

File: sys-apps/ethq/ethq-0.5.0.ebuild [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+	fi
+}
+
+src_compile() {
+	emake CXX="$(tc-getCXX)"
+}
+
+src_test() {
+	local driver
+	for driver in "${S}"/tests/* ; do
```
Comment (mgorny) [inline sys-apps/ethq/ethq-0.5.0.ebuild]: Why do you need to use absolute path here?

File: sys-apps/ethq/ethq-0.5.0.ebuild [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+		"${S}"/ethq_test "$(basename "$driver")" "${driver}" \
```
Comment (mgorny) [inline sys-apps/ethq/ethq-0.5.0.ebuild]: Don't call `basename`, use builtin bash `${...##*/}` operation to get the basename.

File: sys-apps/ethq/ethq-0.5.0.ebuild (hunk 1 above)
Comment (SpiderX) [reply sys-apps/ethq/ethq-0.5.0.ebuild]: Replaced.

File: sys-apps/ethq/ethq-0.5.0.ebuild (hunk 2 above)
Comment (SpiderX) [reply sys-apps/ethq/ethq-0.5.0.ebuild]: Used.

File: sys-apps/ethq/ethq-0.5.0.ebuild (hunk 4 above)
Comment (SpiderX) [reply sys-apps/ethq/ethq-0.5.0.ebuild]: Changed to relative.

File: sys-apps/ethq/ethq-0.5.0.ebuild (hunk 5 above)
Comment (SpiderX) [reply sys-apps/ethq/ethq-0.5.0.ebuild]: Replaced.

**Comment by SpiderX** [comment]
Upstream doesn't use any build configuration.
Will something like this enough? :

        if has_version sys-libs/ncurses:0[tinfo] ; then
                sed -i '/LIBS_CURSES/s/-lncurses/-lncurses -ltinfo/' Makefile \
                        || die "sed failed for tinfo"
        fi

It works.

**Comment by mgorny** [comment]
I think it would be cleaner to use pkg-config. Maybe you could override it without sed, e.g.:

    emake LIBS_CURSES="$($(tc-getPKG_CONFIG) --libs ncurses)"

**Comment by mgorny** [comment]
Hmm, there's also `tinfo` package but I don't know if it's present in `ncurses[-tinfo]`. If you have it installed, could you check?

**Comment by SpiderX** [comment]
> Hmm, there's also tinfo package but I don't know if it's present in ncurses[-tinfo].

It is not present. Cloud be used.

Testes with `ncurses[-tinfo]` and with `ncurses[tinfo]` — works.

**Pull request closed without merging**